pysipp/launch.py

```diff
@@ -149,18 +150,19 @@ def clear(self):
         self._procs.clear()
 
 
-async def run_all_agents(runner, agents, timeout=180, block=True):
+async def run_all_agents(runner, agents, timeout, block=True):
     """Run a sequencec of agents using a ``TrioRunner``."""
 
     try:
         await runner.run((ua.render() for ua in agents), timeout=timeout)
         if block:
-            await finalize(runner, agents, timeout)
-        return runner
+            return await finalize(runner, agents, timeout)
+        else:
+            return finalizer(runner, agents)
```

Comment:
Yeah, I just don't see how this is beneficial.

Instead of returning the runner now we're returning a closure that calls `trio.run()` and that closure seems mostly to be a function created solely for the purposes of having it *be* a function (aka an object with `.__call__()`)

I actually find this somewhat hard to follow and not any more *functional* then the original method which does this same thing.

Reply:
What is the benefit of exposing an implementation detail like the runner to the user? The downside is that it leaves most space for the user to misuse the API.